beeai/issue_fetcher/jira_issue_fetcher.py

```diff
@@ -0,0 +1,381 @@
+        """
+        response = requests.post(
+            url,
+            json=json_data,
+            headers=self.headers,
+            timeout=self.API_TIMEOUT
+        )
+                
+        # Handle rate limiting specifically
+        if response.status_code == 429:
+            logger.warning("Rate limited (429), will retry with backoff")
+            raise requests.HTTPError("Rate limited", response=response)
+                
+        response.raise_for_status()
+        return response.json()
+    
+    async def search_issues(self) -> List[Dict[str, Any]]:
+        """
+        Search for issues using the configured query with pagination
+        """
+        logger.info(f"Starting issue search with query: {self.query}")
+        
+        all_issues = []
+        start_at = 0
+        total_issues = None
+        
+        fields = [
+            "key",        # Issue key (e.g., RHEL-12345)
+            "labels",     # Issue labels 
+        ]
+        
+        while True:
+            await self._rate_limit()
+            
+            # Use POST with JSON payload instead of GET with params to handle long queries
+            json_payload = {
+                "jql": self.query,
+                "startAt": start_at,
+                "maxResults": self.max_results_per_page,
+                "fields": fields
+            }
+            
+            logger.info(f"Fetching issues: startAt={start_at}, maxResults={self.max_results_per_page}")
+            
+            try:
+                url = urljoin(self.jira_url, "rest/api/2/search")
+                response_data = self._make_request_with_retries(url, json_data=json_payload)
+                
+                issues = response_data.get("issues", [])
+                all_issues.extend(issues)
+                
+                if total_issues is None:
+                    total_issues = response_data.get("total", 0)
+                    logger.info(f"Total issues found: {total_issues}")
+                
+                logger.info(f"Retrieved {len(issues)} issues (total so far: {len(all_issues)}/{total_issues})")
+                
+                if len(all_issues) >= total_issues or len(issues) == 0:
+                    break
+                
+                start_at += self.max_results_per_page
+                
+            except Exception as e:
+                logger.error(f"Error fetching issues: {e}")
+                raise
+        
+        logger.info(f"Successfully retrieved {len(all_issues)} issues")
+        return all_issues
+    
+    @asynccontextmanager
+    async def redis_client(self) -> AsyncGenerator[redis.Redis, None]:
+        client = redis.Redis.from_url(self.redis_url)
+        try:
+            await client.ping()
+            logger.info("Connected to Redis")
+            yield client
+        finally:
+            await client.aclose()
+            logger.info("Disconnected from Redis")
```

Comment:
This is already in `agents.utils` and could be also moved to `common`:

https://github.com/packit/ai-workflows/blob/67073b91373228cfc19805512f7c12e1e17f55d7/beeai/agents/utils.py#L89-L96